pytorch_lightning/trainer/trainer.py

```diff
@@ -913,8 +913,11 @@ def tune(
 
     def _restore_modules_and_callbacks(self) -> None:
         # restore modules after setup
-        if self.state.fn == TrainerFn.FITTING:
-            self.checkpoint_connector.resume_start()
+        self.checkpoint_connector.resume_start()
+
+        if self.state.fn != TrainerFn.FITTING:
+            return
+
```

Comment:
why this swap?

Reply:
https://github.com/PyTorchLightning/pytorch-lightning/pull/9413#discussion_r711204473